imap/jmap_mailbox.c

```diff
@@ -2538,6 +2544,28 @@ static void _mbox_update(jmap_req_t *req, struct mboxset_args *args,
     }
     ptrarray_append(&strpool, oldmboxname);
 
+    /* Must not set role on inbox */
+    if (args->specialuse && is_inbox) {
+        jmap_parser_invalid(&parser, "role");
+        goto done;
+    }
+
+    /* Skip role updates in first iteration */
+    if (mode == _MBOXSET_SKIP) {
+        if (args->specialuse && *args->specialuse) {
+            result->skipped = 1;
+        }
+        if (!result->skipped) {
+            struct buf val = BUF_INITIALIZER;
+            annotatemore_lookup(mbentry->name, "/specialuse", req->accountid, &val);
+            if (buf_len(&val)) {
+                buf_free(&val);
+                result->skipped = 1;
+            }
```

Comment:
I'd probably write this as:

```
    if (buf_len(&val))
        result->skipped = 1;
    buf_free(&val);
```

The assumption is that if the length is value, nothing was malloced, but that may not always be true!  There's probably heaps of other code that would need reviewing, but it's nice not to add this assumption in more places.

Reply:
done